rma_sale/wizard/sale_order_rma_wizard.py

```diff
@@ -0,0 +1,165 @@
+        self.ensure_one()
+        rma = self.create_rma()
+        rma.action_confirm()
+        action = self.env.ref('rma.rma_action').read()[0]
+        if len(rma) > 1:
+            action['domain'] = [('id', 'in', rma.ids)]
+        elif rma:
+            action.update(
+                res_id=rma.id,
+                view_mode="form",
+                view_id=False,
+                views=False,
+            )
+        return action
+
+
+class SaleOrderLineRmaWizard(models.TransientModel):
+    _name = "sale.order.line.rma.wizard"
+    _description = 'Sale Order Line Rma Wizard'
+
+    wizard_id = fields.Many2one(
+        comodel_name='sale.order.rma.wizard',
+        string='Wizard',
+    )
+    order_id = fields.Many2one(
+        comodel_name='sale.order',
+        related='wizard_id.order_id',
+    )
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product',
+        required=True,
+    )
+    uom_category_id = fields.Many2one(
+        comodel_name='uom.category',
+        related='product_id.uom_id.category_id',
+    )
+    quantity = fields.Float(
+        string='Quantity',
+        digits=dp.get_precision('Product Unit of Measure'),
+        required=True,
+    )
+    uom_id = fields.Many2one(
+        comodel_name='uom.uom',
+        string='Unit of Measure',
+        domain="[('category_id', '=', uom_category_id)]",
+        required=True,
+    )
+    picking_ids = fields.Many2many(
+        comodel_name='stock.picking',
+        compute='_compute_picking_ids'
+    )
+    picking_id = fields.Many2one(
+        comodel_name='stock.picking',
+        string='Delivery order',
+        domain="[('id', 'in', picking_ids)]"
+    )
+    move_id = fields.Many2one(
+        comodel_name='stock.move',
+        compute='_compute_move_id'
+    )
+    operation_id = fields.Many2one(
+        comodel_name='rma.operation',
+        string='Requested operation',
+    )
+
+    @api.onchange('move_id')
+    def _compute_move_id(self):
+        for record in self:
+            if record.picking_id:
+                record.move_id = record.picking_id.move_lines.filtered(
+                    lambda r: (r.sale_line_id.product_id == record.product_id
+                               and r.sale_line_id.order_id == record.order_id))
+
+    @api.onchange('product_id')
+    def _compute_picking_ids(self):
+        for record in self:
+            order_line = record.order_id.order_line.filtered(
+                lambda r: r.product_id == record.product_id)
+            record.picking_ids = order_line.mapped('move_ids.picking_id')
+
+    @api.model
+    def fields_get(self, allfields=None, attributes=None):
+        res = super().fields_get(allfields, attributes=attributes)
+        order_id = self.env.context.get('active_id')
+        order = self.env['sale.order'].browse(order_id)
+        res['product_id']['domain'] = [
+            ('id', 'in', order.order_line.mapped('product_id').ids)
```

Comment:
Why do it this way? Use a domain with a computed field `allowed_product_ids` similar to current `allowed_picking_ids`

Reply:
This is still pending